dangerzone/args.py

```diff
@@ -17,9 +18,46 @@ def validate_input_filename(
     return filename
 
 
+def sanitize_parameters(original_param: str) -> str:
+    """
+    Mitigates command injection vulnerabilities. Checks if an argument exists
+    could have originated from a file
+
+    Example without mitigation:
+
+        $ dangerzone *
+
+    One file is maliciously named "--ocr-lang" or some other parameter would be
+    interpreted as a parameter and not as a file.
+    """
+    parameter = f"--{original_param}"
```

Comment:
I think this doesn't take into account single dash arguments (`-v`), concatenated single dash arguments (`-aux`), or double dash arguments with an equal sign (`--output-filename=sample.pdf`). I think it's best to consult `sys.argv` instead:

1. List the files in the current working directory, and create a `set`.
2. Filter the args in `sys.argv` that start with `-`, and create a `set`.
3. Ensure that there is no intersection between these sets. Else, report the offending arguments.

This can take place after Click has parsed the arguments, we don't need to hook into its internals I believe.

Reply:
Good catch.

I am struggling with some interesting friction with the `click` module. It's not trivial do to stuff outside of click:
  - **doing the parameter check in the `cli_main`** means `click` already did its own parsing and could have interpreted arguments like `--help`.
  - **doing it wrapping the `cli_main`** (before the click wrappers) makes the whole function a non-click command, which then leads to issues when testing using click's `CliRunner`. For example, when running `pytest tests/test_cli.py::TestSecurity` we get:
    ```python
    [...]
    E       assert 'Security: one file is maliciously named' in ''
    E        +  where '' = <CLIResult AttributeError("'function' object has no attribute 'main'")>.stdout
    ```

Furthermore I found that we could use a click function to detect parameters from `sys.argv`. A simple function but not worth reinventing the wheel if we are already depending on click. The function is `prefix, parameter = click.parser.split_opt(arg)`.

So it is a catch-22 situation. I have a working code to do what you proposed, but I don't have a way execute it before click without breaking the tests. If I do it like I was, with `token_normalize_func` trick the code run over every argument. So getting it from `sys.argv` every time doesn't make sense.

Way too much time spent on this already. I will put a pause on this one for now.

Reply:
Yeah, I get your pain. I hit against the same wall, and I had to override a public Click method to achieve CLI and test parity. It's in the `209-bulk-doc-cli-support-2` branch.

Reply:
I see now that you've found a solution to this https://github.com/freedomofpress/dangerzone/commit/44e32bd239778e3461d0f5f3a265a21f22242190#diff-1818f923cb4b1458022430fcaea85123d0a5d8fd15b270705b5ba7ba085938deR89